Userland/Libraries/LibTest/Randomized/ShrinkCommand.h

```diff
@@ -0,0 +1,247 @@
+            [](DeleteChunkAndMaybeDecPrevious c) { return String::formatted("DeleteChunkAndMaybeDecPrevious({})", c.chunk); },
+            [](MinimizeChoice c) { return String::formatted("MinimizeChoice(i={})", c.index); },
+            [](RedistributeChoicesAndMaybeInc c) { return String::formatted("RedistributeChoicesAndMaybeInc(left={},right={})", c.left_index, c.right_index); },
+            [](SwapChunkWithNeighbour c) { return String::formatted("SwapChunkWithNeighbour({})", c.chunk); });
+    }
+
+    template<typename C1, typename C2, typename C3, typename C4, typename C5, typename C6>
+    auto visit(C1 on_zero, C2 on_sort, C3 on_delete, C4 on_minimize, C5 on_redistribute, C6 on_swap_chunk)
+    {
+        return m_cmd.visit(
+            [&](ZeroChunk c) { return on_zero(c); },
+            [&](SortChunk c) { return on_sort(c); },
+            [&](DeleteChunkAndMaybeDecPrevious c) { return on_delete(c); },
+            [&](MinimizeChoice c) { return on_minimize(c); },
+            [&](RedistributeChoicesAndMaybeInc c) { return on_redistribute(c); },
+            [&](SwapChunkWithNeighbour c) { return on_swap_chunk(c); });
+    }
+
+private:
+    CommandVariant m_cmd;
+
+    // Will generate ShrinkCommands for all chunks of sizes 1,2,3,4,8 in bounds of the
+    // given RandomRun size.
+    //
+    // They will be given in a reverse order (largest chunks first), to maximize our
+    // chances of saving work (minimizing the RandomRun faster).
+    //
+    // chunkCmds(10, false, [](Chunk c){ return SortChunk(c); })
+    // -->
+    // [ // Chunks of size 8
+    //   SortChunk { chunk_size = 8, start_index = 0 }, // [XXXXXXXX..]
+    //   SortChunk { chunk_size = 8, start_index = 1 }, // [.XXXXXXXX.]
+    //   SortChunk { chunk_size = 8, start_index = 2 }, // [..XXXXXXXX]
+    //
+    //   // Chunks of size 4
+    //   SortChunk { chunk_size = 4, start_index = 0 }, // [XXXX......]
+    //   SortChunk { chunk_size = 4, start_index = 1 }, // [.XXXX.....]
+    //   // ...
+    //   SortChunk { chunk_size = 4, start_index = 5 }, // [.....XXXX.]
+    //   SortChunk { chunk_size = 4, start_index = 6 }, // [......XXXX]
+    //
+    //   // Chunks of size 3
+    //   SortChunk { chunk_size = 3, start_index = 0 }, // [XXX.......]
+    //   SortChunk { chunk_size = 3, start_index = 1 }, // [.XXX......]
+    //   // ...
+    //   SortChunk { chunk_size = 3, start_index = 6 }, // [......XXX.]
+    //   SortChunk { chunk_size = 3, start_index = 7 }, // [.......XXX]
+    //
+    //   // Chunks of size 2
+    //   SortChunk { chunk_size = 2, start_index = 0 }, // [XX........]
+    //   SortChunk { chunk_size = 2, start_index = 1 }, // [.XX.......]
+    //   // ...
+    //   SortChunk { chunk_size = 2, start_index = 7 }, // [.......XX.]
+    //   SortChunk { chunk_size = 2, start_index = 8 }, // [........XX]
+    // ]
+    template<typename FN>
+    static Vector<ShrinkCommand> chunk_cmds(size_t run_size, bool allow_chunks_size1, FN chunk_to_cmd)
```

Comment:
two comments here: 

1. can we not be afraid of the word "commands" in the function names?
2. prefer enum classes to bool parameters. e.g. 

```
enum class AllowSizeOneChunks {
   Yes,
   No,
};
```
called like
```
static Vector<ShrinkCommand> deletion_commands(size_t run_size)
    {
        return chunk_commands(
            run_size,
            AllowSizeOneChunks::Yes,
            [](Chunk c) { return ShrinkCommand(DeleteChunkAndMaybeDecPrevious { c }); });
    }
 ```